### PR TITLE
Adding service listing and execution API

### PR DESCRIPTION
**Pull request by kumarashit**
This is the connector API for listing the actions and executing action.
Here is the test and coverage result:
============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.12, pytest-4.4.1, py-1.8.0, pluggy-0.9.0
rootdir: /root/go/src/github.com/opensds/orchestration, inifile: pytest.ini
plugins: cov-2.6.1
collected 2 items

tests/unit/api/test_instances.py .                                                                                                         [ 50%]
tests/unit/api/test_services.py .                                                                                                          [100%]

---------- coverage: platform linux2, python 2.7.12-final-0 ----------
Name                                                        Stmts   Miss Branch BrPart  Cover
---------------------------------------------------------------------------------------------
orchestration/__init__.py                                       0      0      0      0   100%
orchestration/api/__init__.py                                   0      0      0      0   100%
orchestration/api/instances.py                                 12      0      0      0   100%
orchestration/api/services.py                                  10      0      0      0   100%
orchestration/api/tasks.py                                      0      0      0      0   100%
orchestration/connectionmanager/Connector.py                   10      0      2      1    92%
orchestration/connectionmanager/OrchestrationConstants.py      13      1      0      0    92%
orchestration/connectionmanager/St2.py                         39      6      4      1    84%
orchestration/connectionmanager/__init__.py                     0      0      0      0   100%
orchestration/db/__init__.py                                    0      0      0      0   100%
orchestration/db/models.py                                      7      7      0      0     0%
orchestration/server.py                                        19      2      2      1    86%
---------------------------------------------------------------------------------------------
TOTAL                                                         110     16      8      3    84%

============================================================ 2 passed in 0.29 seconds ============================================================

### CODE REVIEW

**Comment by wisererik** [comment]
@kumarashit please fix CI errors

File: orchestration/api/services.py [hunk 1]
```diff
@@ -14,11 +14,15 @@
 
 from flask import jsonify
 from flask import Blueprint
+from orchestration.connectionmanager.Connector import Connector
 # from flask import request
```
Comment (wisererik) [inline orchestration/api/services.py]: remove unused library

File: tests/unit/api/test_instances.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_post_instance(client):
+    # response = client.post('/v1/orchestration/instances')
```
Comment (wisererik) [inline tests/unit/api/test_instances.py]: remove unused code or add unit test.

File: tests/unit/api/test_services.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 
 def test_get_service(client):
-    response = client.get('/v1/orchestration/services/testing')
-    assert response.status_code == 200
+    # response = client.get('/v1/orchestration/services/testing')
```
Comment (wisererik) [inline tests/unit/api/test_services.py]: same from previous comment

File: orchestration/api/services.py (hunk 1 above)
Comment (kumarashit) [reply orchestration/api/services.py]: Done

File: tests/unit/api/test_instances.py (hunk 2 above)
Comment (kumarashit) [reply tests/unit/api/test_instances.py]: Plan to add unit test. Have kept it commented as of now

File: orchestration/api/instances.py [hunk 4]
```diff
@@ -11,3 +11,65 @@
...
+
+instance = Blueprint("instance", __name__)
+
+# API to execute a service.
+# For example, St2 execution of this API will require 'action'
+# as one of the parameter. 'parameters' maybe required for the Actions
+# St2 example of data:/v1/orchestration/instances -H
+# "Content-Type:application/json" -d
+# '{"action":"aks.ak_echo_py", "parameters":{"message":"Hello World"}}'
+@instance.route("/v1/orchestration/instances", methods=['POST'])
```
Comment (xing-yang) [inline orchestration/api/instances.py]: All our other APIs are still at v1beta.  This one should also be v1beta.

File: orchestration/db/__init__.py [hunk 5]
```diff
@@ -19,26 +19,32 @@
...
 def drop_db():
     Base.metadata.drop_all(bind=__engine__)
+
+
+def create_defs():
+    pass
+#    service_def = {}
+#    service_def['name'] = 'migration_st2'
+#    service_def['description'] = 'Container for migration services'
+#    create_service_definition(None, service_def)
```
Comment (xing-yang) [inline orchestration/db/__init__.py]: Remove code that is not needed and commented out.

File: tests/unit/api/test_tasks.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_instance_output(client):
+    # response = client.get('/v1/orchestration/tasks/xyz')
+    # assert response.status_code == 200
```
Comment (xing-yang) [inline tests/unit/api/test_tasks.py]: Remove commented out code.

File: orchestration/db/__init__.py [hunk 7]
```diff
@@ -19,26 +19,32 @@
 from sqlalchemy import engine_from_config
 from sqlalchemy.orm import sessionmaker
 from orchestration.utils.config import DATABASE
-from orchestration.db.models import Base
+from orchestration.db.models import Base, ServiceDefinition
 
 
 # Declares global engine in here
 __engine__ = None
```
[the remark this answers is not in the document]
Comment (kumarashit) [reply orchestration/db/__init__.py]: Fixed

File: orchestration/api/instances.py [hunk 8]
```diff
@@ -11,3 +11,65 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from flask import jsonify
+from flask import Blueprint
+from orchestration.connectionmanager.Connector import Connector
```
[the remark this answers is not in the document]
Comment (kumarashit) [reply orchestration/api/instances.py]: Done

File: orchestration/db/__init__.py (hunk 5 above)
Comment (kumarashit) [reply orchestration/db/__init__.py]: Removed.
Thanks

File: orchestration/api/instances.py (hunk 4 above)
Comment (kumarashit) [reply orchestration/api/instances.py]: changed to v1beta

File: tests/unit/api/test_tasks.py (hunk 6 above)
Comment (kumarashit) [reply tests/unit/api/test_tasks.py]: removed

File: orchestration/api/instances.py [hunk 9]
```diff
@@ -11,3 +11,92 @@
...
+    wf_hash['name'] = ret_json['action']['name']
+    wf_hash['input'] = json.dumps(ret_json['parameters'])
+    wf_hash['workflow_definition_id'] = ret_json['action']['ref']
+
+    # Create the record of this instance in DB
+    create_workflow(None, wf_hash)
+    return jsonify(response=ret_json), 200
+
+
+@instance.route("/v1beta/orchestration/instances/create", methods=['POST'])
```
Comment (himanshuvar) [inline orchestration/api/instances.py]: Can we remove the create from the endpoint

Comment (kumarashit) [reply orchestration/api/instances.py]: Will do that. This api is redundant as of now as we are registering actions through CLI.
Right now code doesn’t get called

Comment (kumarashit) [reply orchestration/api/instances.py]: We can open issue and fix it.

**Pull request merged**